src/core/viz/expressions/Animation.js

```diff
@@ -93,7 +96,46 @@ export class Animation extends BaseExpression {
         return !this.paused;
     }
 
+    _dataReady() {
```

Comment:
It takes some time to understand this logic 😅 and I'm afraid it'll take some time each time, but it seems right. I think I'd break it into some functions to try to make clear the intent of each piece, but maybe that could end up worse... so do as you please